### PR TITLE
Add `llm/mcloud` folder for MosaicML Cloud examples

### PR DESCRIPTION
**Pull request by abhi-mosaic**
This PR adds some MCLI examples and a short README, under `examples/llm/mcloud`.

Still working on some Python SDK examples too...

### CODE REVIEW

File: llm/mcloud/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Using MosaicML Cloud to train LLMs
+
+This folder contains some examples of how to use [MosaicML Cloud](https://www.mosaicml.com/cloud) to launch LLM training runs.
```
Comment (dakinggg) [inline llm/mcloud/README.md:3]:
```suggestion
This folder contains examples of how to use [MosaicML Cloud](https://www.mosaicml.com/cloud) to launch LLM training runs.
```

File: llm/mcloud/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+Full documentation on MosaicML Cloud can be found here: https://mcli.docs.mosaicml.com/.
```
Comment (dakinggg) [inline llm/mcloud/README.md:5]:
```suggestion
Full documentation on MosaicML Cloud can be found at https://mcli.docs.mosaicml.com/.
```

File: llm/mcloud/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+## Using MCLI to launch runs
```
Comment (dakinggg) [inline llm/mcloud/README.md:7]:
```suggestion
## Using MosaicML Command Line Interface (MCLI) to launch runs
```

File: llm/mcloud/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+In this folder, we provide two MCLI examples, [`mcli-1b.yaml`](./mcli-1b.yaml) and [`mcli-1b-custom.yaml`](./mcli-1b-custom.yaml) that demonstrate how to configure and launch training runs using our command-line tool, `mcli`.
+
+The first example, `mcli-1b.yaml`, is minimal, and simply clones this repo (https://github.com/mosaicml/examples), checks out a particular tag, and runs the `main.py` training script. The workload configuration is read from a YAML sitting in the repo (`yamls/mosaic_gpt/1b.yaml`).
```
Comment (dakinggg) [inline llm/mcloud/README.md:11]:
```suggestion
The first example, `mcli-1b.yaml`, simply clones this repo (https://github.com/mosaicml/examples), checks out a particular tag, and runs the `main.py` training script. The workload configuration is read from a YAML sitting in the repo (`yamls/mosaic_gpt/1b.yaml`).
```

File: llm/mcloud/mcli-1b.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  - integration_type: git_repo
+    git_repo: mosaicml/examples
+    git_branch: v0.0.2
+    pip_install: -r llm/requirements.txt
+
+command: |
+  cd examples/llm
+  python convert_c4.py --out_root ./my-copy-c4 --splits val
+  composer main.py yamls/mosaic_gpt/1b.yaml \
+    train_loader.dataset.split=val \
```
Comment (vchiley) [inline llm/mcloud/mcli-1b.yaml:11]: add comment saying this is for demos and that people should set up datasets and point to real datasets for real runs.

File: llm/mcloud/mcli-1b.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+integrations:
+  - integration_type: git_repo
+    git_repo: mosaicml/examples
+    git_branch: v0.0.2
```
Comment (dakinggg) [inline llm/mcloud/mcli-1b.yaml:4]: does this work for tags? There isn't actually a branch with this name

File: llm/mcloud/mcli-1b.yaml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+run_name: mosaic-gpt-1b-gpus-8
+
+gpu_num: 8
+gpu_type: a100_40gb
+cluster: r0z0 # replace with your cluster here!
+
+# The below is injected as a YAML file: /mnt/config/parameters.yaml
+# but is not used in this example.
+parameters: {}
```
Comment (vchiley) [inline llm/mcloud/mcli-1b.yaml:26]: new line

File: llm/mcloud/README.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+The second example, `mcli-1b-custom.yaml`, demonstrates how to inject a custom YAML at runtime (`/mnt/config/parameters.yaml`) and instead pass that file to `main.py`. This enables users to quickly customize training runs without needing to check in their edits to the repository.
+
+Here's how easy it is to launch an LLM training run with MCLI:
+```bash
+mcli run -f mcli-1b.yaml --cluster CLUSTER --gpus GPUS --name NAME --follow
+```
+
+All the details of multi-gpu and multi-node orchestration are handled automatically by MosaicML Cloud. Try it out yourself!
```
Comment (vchiley) [inline llm/mcloud/README.md:20]: new line

**Review by dakinggg — APPROVED** [review]
LGTM.

File: llm/mcloud/README.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+## Using the Python API to launch runs
```
Comment (vchiley) [inline llm/mcloud/README.md:21]: Maybe just something like:
```suggestion
You can also use the [python sdk](https://mcli.docs.mosaicml.com/en/stable/python/hello_world.html#hello-world) to programmatically sweep parameter settings and launch runs
```

**Review by vchiley — APPROVED** [review]
few small comments, but generally lgtm

File: llm/mcloud/mcli-1b.yaml (hunk 6 above)
Comment (abhi-mosaic) [reply llm/mcloud/mcli-1b.yaml:4]: Yup it works!

**Pull request merged**